### PR TITLE
Add KEDAScalersInfo to display important information

### LINKED ISSUE #6328: New eventreason KEDAScalersInfo to display important information
### Proposal

Adding a new event (KEDAScalersInfo) so that important information is shown in e.g. Argo CD.

The result in ArgoCD will be:

image001

### Use-Case

In our organization, we work with multiple SRE teams. The rollout of scaledObjects is done within the SRE teams themselves with Helm and ArgoCD. Whether the rollout is going well is mainly viewed via ArgoCD (observability) itself (Events).
There are important information messages that are missed because of this, such as a parameter that will be deprecated and the info message it no longer working in a certain future version.

### Is this a feature you are interested in implementing yourself?

Yes

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by rickbrouwer**
Adding a new event (KEDAScalersInfo) so that important information is shown in e.g. Argo CD.
Added the deprecation info message by the `cpu_memory_scaler` and `ibmmq_scaler`.

### Checklist

- [X] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [X] Tests have been added
- [X] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #6328

### CODE REVIEW

**Review by wozniakjan — COMMENTED** [review]
would it make sense to include these events in the `TypedConfig` deprecated field?
https://github.com/kedacore/keda/blob/833244628a78b6b21f0ef232e288d20829438fab/pkg/scalers/scalersconfig/typed_config.go#L102

I can imagine we may want to add something like `"DeprecatedEvent"` parameter to model this behaviour, because iirc the `"Deprecated"` parameter present there will error the scaler parsing. Or maybe we change the current behaviour to not fail and throw event insted?

File: pkg/scalers/cpu_memory_scaler.go [hunk 1]
```diff
@@ -58,7 +66,6 @@ func parseResourceMetadata(config *scalersconfig.ScalerConfig, logger logr.Logge
 
 	// This is deprecated and can be removed later
 	if meta.Type != "" {
-		logger.Info("The 'type' setting is DEPRECATED and will be removed in v2.18 - Use 'metricType' instead.")
```
[the remark this answers is not in the document]
Comment (zroubalik) [reply pkg/scalers/cpu_memory_scaler.go:61]: I think it make sense to keep both

[the remark this answers is not in the document]
Comment (rickbrouwer) [reply pkg/scalers/cpu_memory_scaler.go:61]: I have changed the code. I’ve removed the info message directly from the scaler, and it now also runs through `TypedConfig`, so the message only needs to be specified once. So there is an info message and an Event.

**Review by zroubalik — COMMENTED** [review]
> Thank you for the good feedback. In retrospect, it is indeed much better to handle this through `TypedConfig`. I’ve restructured it accordingly. I created a new parameter called `deprecatedAnnounce`. This will issue both, an info message (to maintain the 'old' behavior), and an event. This means I’ve removed the info message directly from the scaler, and it now also runs through `TypedConfig`, so the message only needs to be specified once.
> 
> I believe the other 'deprecated' parameter should remain in place. It makes the flow more logical. First, there will be an announcement, typically for two releases. After that, it can be converted to deprecated status, triggering an error. This is important because it provides the reason for the scaler's failure if the parameter is still in use.
> 
> Perhaps it would be good to set a standard, once something is truly deprecated and causes an error, we keep showing it for two releases and then remove the parameter to keep the code clean. So in summary:
> 
> show 2 releases: `deprecatedAnnounce`: parameter x is deprecated and will be removed in release x then remove the `deprecatedAnnounce`, then; show 2 releases: `deprecated`: parameter x is deprecated, use parameter x instead then remove `deprecated` tag
> 
> What are your thoughts?

I like this :)

We should document this here: https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md

**Review by wozniakjan — APPROVED** [review]
> What are your thoughts?

this is very well thought through from all angles, lgtm

**Comment by rickbrouwer** [comment]
> I like this :)
> 
> We should document this here: https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md

That seems like a good plan to me. I will document this there and offer it via a PR

File: CHANGELOG.md [hunk 2]
```diff
@@ -89,6 +89,7 @@ New deprecation(s):
 ### Other
 
 - **General**: Bump newrelic-client-go deps to 2.51.2 (latest) ([#6325](https://github.com/kedacore/keda/pull/6325))
+- **General**: New eventreason KEDAScalersInfo to display important information ([#6328](https://github.com/kedacore/keda/pull/6328))
```
Comment (zroubalik) [inline CHANGELOG.md:92]: Nit, can we please reference an issue here?

**Review by zroubalik — COMMENTED** [review]
Looking good, thanks!

File: CHANGELOG.md (hunk 2 above)
Comment (rickbrouwer) [reply CHANGELOG.md]: fixed

**Pull request merged**